src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -74,20 +78,25 @@ public Class getType() {
     @Override
     public Object instantiate() throws ConstructionException {
         // find constructor and build up necessary constructor arguments
+
+
         Constructor<?> ctor = getConstructor();
-        Object[] ctorArgs = new Object[ctor.getParameterTypes().length];
-        for (Desire d: desires) {
-            if (d.getInjectionPoint() instanceof ConstructorParameterInjectionPoint) {
-                // this desire is a constructor argument so create it now
-                Instantiator provider = providers.get(d);
-                ConstructorParameterInjectionPoint cd = (ConstructorParameterInjectionPoint) d.getInjectionPoint();
-                ctorArgs[cd.getParameterIndex()] = checkNull(cd, provider.instantiate());
-            }
-        }
-        
         // create the instance that we are injecting
+        LogContext mdcContext = LogContext.create();
         Object instance;
         try {
+            mdcContext.put("org.grouplens.grapht.class", ctor.getClass().toString());
+            Object[] ctorArgs = new Object[ctor.getParameterTypes().length];
+            for (Desire d : desires) {
+                if (d.getInjectionPoint() instanceof ConstructorParameterInjectionPoint) {
+                    // this desire is a constructor argument so create it now
+                    Instantiator provider = providers.get(d);
+                    ConstructorParameterInjectionPoint cd = (ConstructorParameterInjectionPoint) d.getInjectionPoint();
+                    logger.trace("Injection point satisfactions in progress {}",cd);
+                    mdcContext.put("org.grouplens.grapht.injectionPoint", cd.toString());
```

Comment:
This is mostly OK - would be better to make a fresh context for the injection point (so that it's cleared before we run the constructor).